module/VuFind/src/VuFind/Controller/ShibbolethLogoutNotificationController.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * Shibboleth Logout Notification API Controller
+ *
+ * PHP version 5
+ *
+ * Copyright (C) The National Library of Finland 2016.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\Controller;
+
+use Zend\Stdlib\ResponseInterface as Response;
+
+/**
+ * Handles Shibboleth back-channel logout notifications.
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class ShibbolethLogoutNotificationController extends AbstractBase
+{
+    use ApiControllerTrait;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->accessPermission = 'access.api.ShibbolethLogoutNotification';
+    }
+
+    /**
+     * GET method handler for the logout handler
+     *
+     * @return Response
+     */
+    public function getAction()
+    {
+        $this->disableSessionWrites();
+        $response = $this->getResponse();
+        $response->getHeaders()->addHeaderLine(
+            'Content-Type', 'application/wsdl+xml'
+        );
+        $response->setContent($this->getWsdl());
+        return $response;
+    }
+
+    /**
+     * POST method handler for the logout handler
+     *
+     * @return Response
+     */
+    public function postAction()
+    {
+        $this->disableSessionWrites();
+        list($uri) = explode('?', $this->getRequest()->getUriString());
+        $soapServer = new \Zend\Soap\Server(
```

Comment:
If we merge this, we should be sure to update #598 since I suspect the more granular Composer dependencies there may need to be expanded to include Zend\Soap support.